lib/ansible/module_utils/facts/network/iscsi.py

```diff
@@ -80,23 +80,32 @@ def collect(self, module=None, collected_facts=None):
                     break
         elif sys.platform.startswith('aix'):
             if module is not None:
-                rc, out, err = module.run_command('/usr/sbin/lsattr -E -l iscsi0 | grep initiator_name')
+                cmd = module.get_bin_path('lsattr', required=True)
+                cmd = cmd + " -E -l iscsi0"
+                rc, out, err = module.run_command(cmd)
                 if out:
-                    iscsi_facts['iscsi_iqn'] = out.split()[1].rstrip()
+                    for line in out.splitlines():
+                        if 'initiator_name' in line:
+                            iscsi_facts['iscsi_iqn'] = line.split()[1].rstrip()
             else:
-                aixcmd = '/usr/sbin/lsattr -E -l iscsi0 | grep initiator_name'
+                aixcmd = '/usr/sbin/lsattr -E -l iscsi0'
```

Comment:
you use get_bin_path above, i would reuse it for all lsattr calls (and do same for the other utilities below)

Reply:
i can't reuse it , since it is different code path where `module.get_bin_path()` is not available  
`if module is not None:`
https://github.com/ansible/ansible/blob/b615561bfa07f93e20c6d4499cb9b9d93c2f0568/lib/ansible/module_utils/facts/network/iscsi.py#L81-L83


and as i mentioned in the PR body, i'm not sure `get_bin_path()` would find /opt/iscsi/bin/iscsiutil on HP-UX (don't have access to test)...